tests/build/build.go

```diff
@@ -79,14 +80,35 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build-ser
 			_, err = f.AsKubeAdmin.HasController.CreateApplication(applicationName, testNamespace)
 			Expect(err).NotTo(HaveOccurred())
 
-			componentName = fmt.Sprintf("%s-%s", "test-component-pac", util.GenerateRandomString(6))
-			pacBranchName = constants.PaCPullRequestBranchPrefix + componentName
+			customDefaultComponentName = fmt.Sprintf("%s-%s-%s", gitPrefix, "test-custom-default", util.GenerateRandomString(6))
+			customBranchComponentName = fmt.Sprintf("%s-%s-%s", gitPrefix, "test-custom-branch", util.GenerateRandomString(6))
+			pacBranchName = constants.PaCPullRequestBranchPrefix + customBranchComponentName
 			componentBaseBranchName = fmt.Sprintf("base-%s", util.GenerateRandomString(6))
 
 			err = f.AsKubeAdmin.CommonController.Github.CreateRef(helloWorldComponentGitSourceRepoName, helloWorldComponentDefaultBranch, helloWorldComponentRevision, componentBaseBranchName)
 			Expect(err).ShouldNot(HaveOccurred())
 
-			defaultBranchTestComponentName = fmt.Sprintf("test-custom-default-branch-%s", util.GenerateRandomString(6))
+			gitlabToken := utils.GetEnv(constants.GITLAB_TOKEN_ENV, "")
+			Expect(gitlabToken).ShouldNot(BeEmpty())
+
+			secretAnnotations := map[string]string{}
+
+			err = createGitlabBuildSecret(f, "gitlab-build-secret", secretAnnotations, gitlabToken)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			secret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pipelines-as-code-secret",
+					Namespace: testNamespace,
+				},
+				Type: v1.SecretTypeOpaque,
+				StringData: map[string]string{
+					"password": gitlabToken,
+				},
+			}
+			_, err = f.AsKubeAdmin.CommonController.CreateSecret(f.UserNamespace, secret)
```

Comment:
Do we need this one? aren't we already creating gitlab-build-secret with `createGitlabBuildSecret` above?